storagenode/storagenodedb/consoledb.go

```diff
@@ -153,6 +302,113 @@ func (db *consoledb) getDailyBandwidthUsed(ctx context.Context, cond string, arg
 	return bandwidthUsedList, nil
 }
 
+// GetDailyDiskSpaceUsageTotal returns daily disk usage summed across all known satellites
+// for provided time range
+func (db *consoledb) GetDailyDiskSpaceUsageTotal(ctx context.Context, from, to time.Time) (_ []console.SpaceUsageStamp, err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	query := db.Rebind(`SELECT SUM(at_rest_total), timestamp 
+							FROM rollup_space_usages
+							WHERE rollup_id IN (
+								SELECT MAX(rollup_id)
+								FROM rollup_space_usages
+								WHERE ? <= timestamp AND timestamp <= ?
+								GROUP BY DATE(timestamp), satellite_id
+							) GROUP BY DATE(timestamp)`)
+
+	rows, err := db.db.QueryContext(ctx, query, from, to)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		err = errs.Combine(err, rows.Close())
+	}()
+
+	var stamps []console.SpaceUsageStamp
+	for rows.Next() {
+		var atRestTotal float64
+		var timeStamp time.Time
+
+		err = rows.Scan(&atRestTotal, &timeStamp)
+		if err != nil {
+			return nil, err
+		}
+
+		stamps = append(stamps, console.SpaceUsageStamp{
+			AtRestTotal: atRestTotal,
+			TimeStamp:   timeStamp,
+		})
+	}
+
+	return stamps, nil
+}
+
+// GetDailyDiskSpaceUsageSatellite returns daily disk usage for particular satellite
+// for provided time range
+func (db *consoledb) GetDailyDiskSpaceUsageSatellite(ctx context.Context, satelliteID storj.NodeID, from, to time.Time) (_ []console.SpaceUsageStamp, err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	query := db.Rebind(`SELECT *
+							FROM rollup_space_usages
+							WHERE rollup_id IN (
+								SELECT MAX(rollup_id) 
+								FROM rollup_space_usages
+								WHERE satellite_id = ?
+								AND ? <= timestamp AND timestamp <= ?
+								GROUP BY DATE(timestamp)
+							)`)
+
+	rows, err := db.db.QueryContext(ctx, query, satelliteID, from, to)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		err = errs.Combine(err, rows.Close())
+	}()
+
+	var stamps []console.SpaceUsageStamp
+	for rows.Next() {
+		var rollupID int64
+		var satellite storj.NodeID
+		var atRestTotal float64
+		var timeStamp time.Time
+
+		err = rows.Scan(&rollupID, &satellite, &atRestTotal, &timeStamp)
+		if err != nil {
+			return nil, err
+		}
+
+		stamps = append(stamps, console.SpaceUsageStamp{
+			RollupID:    rollupID,
+			SatelliteID: satellite,
+			AtRestTotal: atRestTotal,
+			TimeStamp:   timeStamp,
+		})
+	}
+
+	return stamps, nil
+}
+
+// withTx is a helper method which executes callback in transaction scope
+func (db *consoledb) withTx(ctx context.Context, cb func(tx *sql.Tx) error) error {
+	tx, err := db.db.BeginTx(ctx, nil)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if err != nil {
+			err = errs.Combine(err, tx.Rollback())
+		}
+
+		err = tx.Commit()
```

Comment:
this overwrites the previous non-nil error

Reply:
Yeah, forgot to put `return` here